scio-core/src/test/scala/com/spotify/scio/coders/KryoAtomicCoderTest.scala

```diff
@@ -35,11 +36,25 @@ import org.joda.time.Instant
 import scala.jdk.CollectionConverters._
 import scala.reflect.ClassTag
 import org.apache.beam.sdk.testing.CoderProperties
+import org.codehaus.jackson.node.NullNode
 import org.typelevel.scalaccompat.annotation.nowarn
 
 case class RecordA(name: String, value: Int)
 case class RecordB(name: String, value: Int)
 
+case class Pair(name: String, size: Int)
+case class CaseClassWithGenericRecord(name: String, size: Int, record: GenericRecord)
+case class CaseClassWithSpecificRecord(name: String, size: Int, record: TestRecord)
+
+// additional kryo registrar for avro schema null value
+// deserializes null to the singleton instance for schema equality
+@KryoRegistrar
+class TestKryoRegistrar extends IKryoRegistrar {
+
+  override def apply(k: Kryo): Unit =
+    k.forClass[NullNode](new SingletonSerializer(NullNode.getInstance()))
+}
```

Comment:
Apparently, this test suite was not run before.
This registrar is needed for the set so schema equality passes. Without this, kry creates a new `NullNode` instance failing equality checks